src/test/java/itest/UploadCertificateIT.java

```diff
@@ -0,0 +1,89 @@
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package itest;
+
+import java.io.File;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.multipart.MultipartForm;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+public class UploadCertificateIT extends TestBase {
+
+    static final String CERT_NAME = "cert";
+    static final String FILE_NAME = "empty.cer";
+    static final String TRUSTSTORE_CERT = "truststore/" + FILE_NAME;
+    static final String MEDIA_TYPE = "application/pkix-cert";
+
+    @Test
+    public void shouldNotAddMalformedCertToTrustStore() throws Exception {
+
+        CompletableFuture<Integer> uploadRespFuture = new CompletableFuture<>();
+        ClassLoader classLoader = getClass().getClassLoader();
+        File emptyCert = new File(classLoader.getResource(FILE_NAME).getFile());
+        String path = emptyCert.getAbsolutePath();
+
+        MultipartForm form =
+                MultipartForm.create()
+                        .attribute("name", CERT_NAME)
+                        .binaryFileUpload(CERT_NAME, FILE_NAME, path, MEDIA_TYPE);
+
+        webClient
+                .post(String.format("/api/v2/certificates"))
+                .sendMultipartForm(
+                        form,
+                        ar -> {
+                            if (ar.failed()) {
+                                uploadRespFuture.completeExceptionally(ar.cause());
+                                return;
+                            }
+                            HttpResponse<Buffer> result = ar.result();
+                            uploadRespFuture.complete(result.statusCode());
+                        });
+
+        int statusCode = uploadRespFuture.get(REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        MatcherAssert.assertThat(statusCode, Matchers.equalTo(500));
+        MatcherAssert.assertThat(classLoader.getResource(TRUSTSTORE_CERT), Matchers.equalTo(null));
```

Comment:
I don't think this assertion tests what you want it to - the `/truststore` directory that is mounted when you use `smoketest.sh` is not mounted in the integration tests, so this passing is a false positive of sorts. I don't think we want to mount persistent storage to the integration test suite since that would make it even less repeatable, so I would be in favour of simply removing this assertion. Just checking that the backend responds with a 500 is probably enough. If we had a `GET /api/v2/certificates` we could also check that no new certificate is added there, but without that I don't think there is anything reasonable to do in this case.

Reply:
Okay, sounds good